### ISSUE TITLE
`no-floating-promises` error randomly thrown?

### ISSUE THREAD

**Issue by JakobJingleheimer** [oxc-project/tsgolint#495]
package | version
:-- | :--
node | `24.10.0`
@types/node | `24. 1.0` <br /> `24.10.2`
oxfmt | ` 0.17.0`
oxlint | ` 1.32.0`
tsgolint | ` 0. 8.4`

I'm getting a bunch (dozens, maybe hundreds) of failures for `typescript/no-floating-promises` on code that has no promise whatsoever. For example:

```js
function sanitiseInternalUrl(input: URL["href"] | null | undefined): string
```

```
  × typescript-eslint(no-floating-promises): Promises must be awaited.
    ╭─[…/sanitiseInternalUrl.test.ts:39:3]
 38 │     
 39 │ ╭─▶   it('should reject disallowed protocols', () => {
 40 │ │       const js = 'alert("bad")';
 41 │ │       for (const protocol of ['data:', 'javascript:']) {
 42 │ │         const i1 = `${protocol}/login;${js}`;
 43 │ │         assert.equal(sanitiseInternalUrl(i1), '', i1);
 44 │ │   
 45 │ │         const i2 = `${protocol}/${location.host};${js}`;
 46 │ │         assert.equal(sanitiseInternalUrl(i2), '', i2);
 47 │ │       }
 48 │ ╰─▶   });
 49 │     });
    ╰────
```

Possibly related to #482?

**Comment by camc314** [comment]
Hi, thanks for reporting, what testing framework are you using? 

Also note that this is reporting that the `it` call is  not being awaited - the `sanitisdeInternalUrl` calls are OK

**Comment by JakobJingleheimer** [comment]
Perhaps it's because of the type def of `node:test.it`'s `TestFn` callback being defined as returning `void | Promise<void>`?

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2acce86be3627a437647a280a0e05d188be6c282/types/node/test.d.ts#L221

Or because the type def of `node:test.it` itself returns `Promise<void>` (but that doesn't need to be awaited—the test runner handles it, and in many/most cases, awaiting it would be undesirable)

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2acce86be3627a437647a280a0e05d188be6c282/types/node/test.d.ts#L125-L128

**Comment by JakobJingleheimer** [comment]
> what testing framework are you using

🤦‍♂️ sorry, `node:test`

**Comment by camc314** [comment]
yeah this sounds like it's doing what it's meant to. you probably want to configure the rule like this:

```

{
  "@typescript-eslint/no-floating-promises": [
    "error",
    {
      "allowForKnownSafeCalls": [
        {
          "from": "package",
          "name": ["test", "it", "suite", "describe"],
          "package": "node:test"
        }
      ]
    }
  ],
}

```

**Comment by JakobJingleheimer** [comment]
Huzzah! That did it, thanks!

Might tsgolint consider a builtin list of defaults for known exceptions? Otherwise every user of node's test runner (and probably others) will have to make this same config/override.

If not, I could probably find somewhere to document it in node's Learn section (but that will likely not reach users who run into this). Perhaps in oxlint / tsgolint's docs would be better though? (If that's the route you prefer to go, I'll happily send over a PR)

**Comment by camc314** [comment]
> Might tsgolint consider a builtin list of defaults for known exceptions? Otherwise every user of node's test runner (and probably others) will have to make this same config/override.

Probably not, just the reasoning being is where do you draw the line of adding a default or known exception, and then it'll probably only be a matter of time until someome is using the node test API in a really strange way and they want this to be reported.

> Perhaps in oxlint / tsgolint's docs would be better though? (If that's the route you prefer to go, I'll happily send over a PR)

This might make sense? Perhapse more explicit examples on the docs pages?

[https://oxc.rs/docs/guide/usage/linter/rules/typescript/no-floating-promises.html#typescript-no-floating-promises](https://oxc.rs/docs/guide/usage/linter/rules/typescript/no-floating-promises.html#typescript-no-floating-promises)

Thanks for getting to the bottom of this so quickly!

**Comment by JakobJingleheimer** [comment]
> Probably not, just the reasoning being is where do you draw the line of adding a default or known exception

I was thinking that as I was typing my message 😅 but then I thought "the error is actually directing users to do something very likely harmful that will likely cost them real money" (hugely slowing down CI runs). For instance, if I do await the test cases in just 1 of many test files, execution time of that 1 test file goes from 1.08s → 24.1s (parallel→ serial). Extrapolate that against a 58s current execution time for the whole suite becomes… 23 minutes (?).

---

I'll send over today or tomorrow a doc update PR with an explicit note and some more examples 🙂
